cms/core/cache.py

```diff
@@ -36,3 +42,60 @@ def get_default_cache_control_decorator() -> Callable:
     """
     cache_control_kwargs = get_default_cache_control_kwargs()
     return cache_control(**cache_control_kwargs)
+
+
+class ElastiCacheIAMCredentialProvider(redis.CredentialProvider):
+    """A custom redis credential provider to use IAM for authentication.
+
+    https://redis.readthedocs.io/en/stable/examples/connection_examples.html#Connecting-to-a-redis-instance-with-ElastiCache-IAM-credential-provider.
+    """
+
+    # Authentication tokens are only valid for a maximum of 15 minutes.
+    TOKEN_TTL = 900
+
+    def __init__(self, user: str, cluster_name: str, region: str):
+        self.user = user
+        self.cluster_name = cluster_name
+        self.region = region
+
+        session = botocore.session.get_session()
+        self.request_signer = RequestSigner(
+            ServiceId("elasticache"),
+            self.region,
+            "elasticache",
+            "v4",
+            session.get_credentials(),
+            session.get_component("event_emitter"),
+        )
+
+        self.cache_key = f"elasticache_{user}_{cluster_name}_{region}"
+
+        self.connection_url = urlunsplit(
```

Comment:
Minor, let's make any attributes not needed to external callers private.

Reply:
Python doesn't really have private attributes. Prefixing with understores might hint it, but then requires a lot of ignores as part of tests. Only `get_credentials` is part of the public API.

Reply:
I know but we have linters that do enforce it...
But not a blocker.

Reply:
I would also advocate for marking things as "private" in cases like this. While not enforced, it's useful to communicate the usage intent of the attributes in a way that can't be missed by future developers, and can be caught in linting.